src/analysis/utils/app-utils.ts

```diff
@@ -84,24 +84,29 @@ export const getAppStatusForDisplay = (status: LeoAppStatus): DisplayAppStatus =
     ['prestarting', () => 'Resuming'],
     ['prestopping', () => 'Pausing'],
     ['provisioning', () => 'Creating'],
-    [Utils.DEFAULT, () => _.capitalize(status)]
+    // TODO: is this case valid?
+    [Utils.DEFAULT, () => _.capitalize(status) as DisplayAppStatus]
   );
 
 export const getEnvMessageBasedOnStatus = (app: App | undefined): string => {
   const waitMessage = 'This process will take up to a few minutes.';
   const nonStatusSpecificMessage =
     'A cloud environment consists of application configuration, cloud compute and persistent disk(s).';
-  return !app
-    ? nonStatusSpecificMessage
-    : Utils.switchCase(
-        app.status,
-        ['PROVISIONING', () => 'The cloud compute is provisioning, which may take several minutes.'],
-        ['STOPPED', () => 'The cloud compute is paused.'],
-        ['PRESTOPPING', () => 'The cloud compute is preparing to pause.'],
-        ['STOPPING', () => `The cloud compute is pausing. ${waitMessage}`],
-        ['PRESTARTING', () => 'The cloud compute is preparing to resume.'],
-        ['STARTING', () => `The cloud compute is resuming. ${waitMessage}`],
-        ['RUNNING', () => nonStatusSpecificMessage],
-        ['ERROR', () => 'An error has occurred on your cloud environment.']
-      );
+
+  if (!app) {
+    return nonStatusSpecificMessage;
+  }
+
+  const statusMessages: Record<LeoAppStatus, string> = {
+    PROVISIONING: 'The cloud compute is provisioning, which may take several minutes.',
+    STOPPED: 'The cloud compute is paused.',
+    // @ts-expect-error Is PRESTOPPING a valid app status?
+    PRESTOPPING: 'The cloud compute is preparing to pause.',
```

Comment:
Hmm.. actually `Prestoopping` and `prestarting` used to be statuses the backend sent, but now these are sanitized out when they are encoded into json.

Reply:
It also looks like there are several possible values of LeoAppStatus not handled here ("STATUS_UNSPECIFIED", "DELETING", "UPDATING").

Reply:
https://github.com/DataBiosphere/terra-ui/pull/4154/commits/750dfe3f3656b03f328568a5fafd5ddbf0761a57 removes the invalid PRESTOPPING and PRESTARTING values here and updates the return type to `string | undefined` instead of `string`.